src/components/Dashboard/Dashboard.js

```diff
@@ -42,101 +51,93 @@ const dashboardMap = {
     [PRINT_LAYOUT]: <PrintLayoutDashboard />,
 }
 
-export class Dashboard extends Component {
-    setDashboard = () => {
-        if (this.props.dashboardsLoaded) {
-            const id = this.props.match.params.dashboardId || null || null
-
-            this.props.selectDashboard(id)
-
-            this.setHeaderbarVisibility()
-        }
-    }
-
-    setHeaderbarVisibility = () => {
-        const header = document.getElementsByTagName('header')[0]
-        if (isPrintMode(this.props.mode)) {
-            header.classList.add('hidden')
-        } else {
-            header.classList.remove('hidden')
+export const Dashboard = ({
+    id,
+    mode,
+    dashboardsLoaded,
+    dashboardsIsEmpty,
+    routeId,
+    selectDashboard,
+    setWindowHeight,
+}) => {
+    useEffect(() => {
+        setHeaderbarVisibility(mode)
+    }, [mode])
+
+    useEffect(() => {
+        if (dashboardsLoaded && !dashboardsIsEmpty) {
+            selectDashboard(routeId)
         }
-    }
+    }, [dashboardsLoaded, dashboardsIsEmpty, routeId, mode])
```

Comment:
Is `mode` needed as a dependency here?

Reply:
Yes it is. This kind of shows how the components are a little too complex and co-dependent at the moment. But I could look to see if there is a better way.

Reply:
I'm afraid it will involve a bigger refactor of one of the thunks. I'd prefer to wait.

Reply:
What about adding the selectDashboard logic to the above affect (which has a legitimate `mode` dependency) instead?  Otherwise I worry that this might be accidentally removed in a future refactor.

Reply:
The logic here with lots of inter-connected props seems like it could be refactored somehow...

Reply:
@amcgee Your observation led me to dig into why the `mode` prop was needed, and it led me to a TODO in my own code, which I have now fixed. So I was able to remove `mode` after all.  Some additional refactoring was needed, and I am gradually gathering all the print constants into a single place.